src/metabase/query_processor/parameters.clj

```diff
@@ -10,18 +10,116 @@
             [metabase.util :as u])
   (:import (org.joda.time DateTimeConstants DateTime)))
 
+(defn- day-range
+  [^DateTime end ^DateTime start]
+  {:end   end
+   :start start})
 
-(def ^:private ^:const relative-dates
-  #{"today"
-    "yesterday"
-    "past7days"
-    "past30days"
-    "thisweek"
-    "thismonth"
-    "thisyear"
-    "lastweek"
-    "lastmonth"
-    "lastyear"})
+(defn- week-range
+  [^DateTime end ^DateTime start]
```

Comment:
when you have metadata tags for variables we usually separate them by commas,

e.g.

```clojure
(defn- week-range
  [^DateTime end, ^DateTime start]
  ...)
```

for clarity (commas are just considered whitespace in Clojure)